src/mlpack/tests/load_save_test.cpp

```diff
@@ -1727,6 +1731,78 @@ BOOST_AUTO_TEST_CASE(CategoricalNontransposedCSVLoadTest04)
     remove("test.csv");
 }
 
+/**
+ * Test For MinMax Scaler Class.
+ */
+BOOST_AUTO_TEST_CASE(MinMaxScalerTest)
+{
+    arma::mat dataset = "-1 -0.5 0 1;"
+                        "2 6 10 18;";
+    arma::mat scaleddataset;
+    arma::mat temp;
+    arma::mat scaled = "0 0.2500 0.5000 1.000;"
+                       "0 0.2500 0.5000 1.000;";
+    data::MinMaxScaler scale;
+    scale.Transform(dataset, scaleddataset);
+    scale.InverseTransform(scaleddataset, temp);
+    CheckMatrices(scaleddataset, scaled);
+    CheckMatrices(dataset, temp);
+}
+
+/**
+ * Test For MaxAbs Scaler Class.
+ */
+BOOST_AUTO_TEST_CASE(MaxAbsScalerTest)
+{
+    arma::mat dataset = "-1 -0.5 0 1;"
+                        "2 6 10 18;";
+    arma::mat scaleddataset;
+    arma::mat temp;
+    arma::mat scaled = "-1 -0.5 0 1;"
+                       "0.1111111111 0.3333333333 0.55555556 1.0000;";
```

Comment:
In general, mlpack matrices are column-major (because of Armadillo, which is itself column-major because of FORTRAN), meaning that each *column* is one data point.  In the tests you've written here, it looks like row-major is assumed.  So probably the implementations should all be transposed to work on rows (dimensions) instead of columns (points).

Reply:
Umm I am aware of the fact that armadillo is coulmn-major and each column is one data point, and that is why `itemMean = arma::mean(input, 1);` I have take mean row wise

In the test , I have 4 data points with 2 features each, So I guess transposing the matrix wouldn't break my code either because it would simply mean 2 data points with 4 features.

May be I could be wrong, and if it is so, could you please let me know in detail.

Reply:
Ah, sorry, you're right.  I got myself confused.  Sorry about that.